source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -13,43 +13,116 @@
 from comtypes import COMError
 from UIAUtils import isTextRangeOffscreen
 from . import UIATextInfo
-from ..behaviors import KeyboardHandlerBasedTypedCharSupport
+from ..behaviors import EnhancedTermTypedCharSupport
 from ..window import Window
 
 
 class consoleUIATextInfo(UIATextInfo):
-
 	def __init__(self, obj, position, _rangeObj=None):
 		# We want to limit  textInfos to just the visible part of the console.
 		# Therefore we specifically handle POSITION_FIRST, POSITION_LAST and POSITION_ALL.
-		# We could use IUIAutomationTextRange::getVisibleRanges, but it seems very broken in consoles
-		# once more than a few screens worth of content has been written to the console.
-		# Therefore we resort to using IUIAutomationTextPattern::rangeFromPoint
-		# for the top left, and bottom right of the console window.
+		# microsoft/terminal#4495: In newer consoles,
+		# IUIAutomationTextRange::getVisibleRanges returns a reliable contiguous range.
+		visiRange = None
+		try:
```

Comment:
How does this code know whether the fetched visible range is appropriate on older versions of Windows 10?

Reply:
If the console returns more than one range then it's an older version of Windows, so visible ranges are unreliable. Ideally we'd always use visible ranges on `ConsoleUIATextInfo` instances and always rangeFromPoint on `ConsoleUIATextInfoEndInclusive` instances, but I'm not sure how to cleanly do this.

Reply:
That's exactly what I wanted to suggest.
You could always split out stuff to one or more helper methods that are called by __init__.

Reply:
> That's exactly what I wanted to suggest.
> You could always split out stuff to one or more helper methods that are called by **init**.

Where would the separation into separate methods most make sense here? In case of `rangeFromPoint` we could write something that returns the correct `_RangeObj` based on the passed-in position for when `super` is called,but for visible ranges we need to call `collapse` *after* `super` is called.